src/nupic/algorithms/TemporalMemory.cpp

```diff
@@ -663,7 +672,7 @@ UInt TemporalMemory::columnForCell(const CellIdx cell) const {
   return cell / cellsPerColumn_;
 }
 
-vector<CellIdx> TemporalMemory::cellsForColumn(Int column) {
+vector<CellIdx> TemporalMemory::cellsForColumn(Int column) { //TODO remove, incorrect w/o topology
```

Comment:
Off-topic:
@ctrl-z-9000-times TM does not use topology, so how can this be correct?
I would like if topology (and cellsToColumns()) could be moved to Connections, I think it's a generic feature that could be useful on other scenarios. 
Related ColumnPooler #293  (that defines Topology), Anomaly #347 (that needs to convert from cells to columns)

Reply:
The word "columns" is used in two different contexts with two different meanings.

The TM & Anomaly have *mini*-columns.  Mini-Columns are not a generic feature, they should stay where they are (in the TM).

The SP & CP have *macro*-columns, which have to do with large scale topology.

Reply:
> The TM & Anomaly have mini-columns. Mini-Columns are not a generic feature, they should stay where they are (in the TM). 

and a mini-column: 
- uses same feed-forward input, has different horizontal connections to other cells
- uses naive topology: if TM has 8 cells with indices 1..8, and 4 cells-per-column: 
  then cells 1-4 are in mini-col 1, c 5-8 in mini-col 2; 
  - Q1 does it make sense to support more complex topologies? 
  - Q2 do we support nD TM? would it flatten nicely that neighbouring cells (in 3D) would have nearby indices (CellIdx)? 
- VectorHelpers::cellsToColumns() should be then moved to TM, as it makes sense only for TM's cells, not any vector/SDR.

Reply:
Yes, that is correct.

>  Q1 does it make sense to support more complex topologies?

Not really. The TM does not have topology.  It has groups of cells which are organized into mini-columns. The mini-columns don't have any topology between them. The SP -not TM- is responsible for large scale topology.

> Q2 do we support nD TM? would it flatten nicely that neighbouring cells (in 3D) would have nearby indices (CellIdx)?

No.  The TM can work with nD inputs, by flattening them into 1D inputs.  This works as long as the SP has the correct topology and imparts it on the SDR which it gives to the TM.

>  VectorHelpers::cellsToColumns() should be then moved to TM, as it makes sense only for TM's cells, not any vector/SDR.

Yes.  Or better yet mixed with the anomaly code which will be baked into the TM.